internal/pkg/otel/configtranslate/otelconfig_test.go

```diff
@@ -0,0 +1,338 @@
+)
+
+func TestBeatNameToDefaultDatastreamType(t *testing.T) {
+	tests := []struct {
+		beatName      string
+		expectedType  string
+		expectedError error
+	}{
+		{
+			beatName:     "filebeat",
+			expectedType: "logs",
+		},
+		{
+			beatName:     "metricbeat",
+			expectedType: "metrics",
+		},
+		{
+			beatName:      "cloudbeat",
+			expectedError: fmt.Errorf("input type not supported by Otel: "),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("%v", tt.beatName), func(t *testing.T) {
+			comp := component.Component{
+				InputSpec: &component.InputRuntimeSpec{
+					BinaryName: "agentbeat",
+					Spec: component.InputSpec{
+						Command: &component.CommandSpec{
+							Args: []string{tt.beatName},
+						},
+					},
+				},
+			}
+			actualType, actualError := getDefaultDatastreamTypeForComponent(&comp)
+			assert.Equal(t, tt.expectedType, actualType)
+
+			if tt.expectedError != nil {
+				assert.Error(t, actualError)
+				assert.EqualError(t, actualError, tt.expectedError.Error())
+			} else {
+				assert.NoError(t, actualError)
+			}
+		})
+	}
+}
+
+func TestGetSignalForComponent(t *testing.T) {
+	tests := []struct {
+		name           string
+		component      component.Component
+		expectedSignal pipeline.Signal
+		expectedError  error
+	}{
+		{
+			name:          "no input spec",
+			component:     component.Component{InputType: "test"},
+			expectedError: fmt.Errorf("input type not supported by Otel: %s", "test"),
+		},
+		{
+			name: "not agentbeat",
+			component: component.Component{
+				InputType: "test",
+				InputSpec: &component.InputRuntimeSpec{
+					BinaryName: "cloudbeat",
+				},
+			},
+			expectedError: fmt.Errorf("input type not supported by Otel: %s", "test"),
+		},
+		{
+			name: "filebeat",
+			component: component.Component{
+				InputType: "filestream",
+				InputSpec: &component.InputRuntimeSpec{
+					BinaryName: "agentbeat",
+					Spec: component.InputSpec{
+						Command: &component.CommandSpec{
+							Args: []string{"filebeat"},
+						},
+					},
+				},
+			},
+			expectedSignal: pipeline.SignalLogs,
+		},
+		{
+			name: "metricbeat",
+			component: component.Component{
+				InputType: "filestream",
+				InputSpec: &component.InputRuntimeSpec{
+					BinaryName: "agentbeat",
+					Spec: component.InputSpec{
+						Command: &component.CommandSpec{
+							Args: []string{"metricbeat"},
+						},
+					},
+				},
+			},
+			expectedSignal: pipeline.SignalLogs,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualSignal, actualError := getSignalForComponent(&tt.component)
+			assert.Equal(t, tt.expectedSignal, actualSignal)
+
+			if tt.expectedError != nil {
+				assert.Error(t, actualError)
+				assert.EqualError(t, actualError, tt.expectedError.Error())
+			} else {
+				assert.NoError(t, actualError)
+			}
+		})
+	}
+}
+
+func TestGetOtelConfig(t *testing.T) {
+	agentInfo := &info.AgentInfo{}
+	fileStreamConfig := map[string]any{
```

Comment:
You probably want a configuration in this test that includes the "input with streams" syntax, for example:

```yaml
  - type: filestream
    id: input-id-a
    use_output: default
    streams:
      - id: filestream-stream-id-a
        data_stream:
          dataset: generic.first
        paths:
          - /var/log/*.log
      - id: filestream-stream-id-b
        data_stream:
          dataset: generic.second
        paths:
          - /var/log/*.log
```

I tested this and it works because the underlying Agent to Beat configuration parsing handles it, but it doesn't cost much to confirm it here.

Multiple outputs works properly as well I think because the component model is handling the grouping before we even get into this code 👍

Reply:
Good catch, done.